test/clj/rems/application/test_cleaner.clj

```diff
@@ -0,0 +1,80 @@
+                       :application/id app-id
+                       :application/external-id ""
+                       :application/resources []
+                       :application/licenses []
+                       :application/forms []
+                       :workflow/id 0
+                       :workflow/type :workflow/default}
+        draft-saved-event {:event/type :application.event/draft-saved
+                           :event/time date-time
+                           :event/actor actor
+                           :application/id app-id
+                           :application/field-values []}]
+    (events/add-event! created-event)
+    (events/add-event! draft-saved-event)
+    app-id))
+
+(defn- add-dummy-application!
+  [{:keys [date-time actor]}]
+  (let [app-id (:id (db/create-application!))
+        created-event {:event/type :application.event/created
+                       :event/time date-time
+                       :event/actor actor
+                       :application/id app-id
+                       :application/external-id ""
+                       :application/resources []
+                       :application/licenses []
+                       :application/forms []
+                       :workflow/id 0
+                       :workflow/type :workflow/default}
+        draft-saved-event {:event/type :application.event/draft-saved
+                           :event/time date-time
+                           :event/actor actor
+                           :application/id app-id
+                           :application/field-values []}
+        submitted-event {:event/type :application.event/submitted
+                         :event/time date-time
+                         :event/actor actor
+                         :application/id app-id}]
+    (events/add-event! created-event)
+    (events/add-event! draft-saved-event)
+    (events/add-event! submitted-event)
+    app-id))
+
+(deftest cleaner
+  (testing "removes expired draft applications"
+    (let [app-id-1 (add-dummy-draft-application! {:date-time (time/now) :actor "alice"})
+          app-id-2 (add-dummy-application! {:date-time (time/now) :actor "alice"})
+          app-id-3 (add-dummy-application! {:date-time (time/minus (time/now) (time/days 120))
+                                            :actor "alice"})
+          expired-app-id (add-dummy-draft-application! {:date-time (time/minus (time/now) (time/days 90) (time/seconds 1))
+                                                        :actor "alice"})
+          expired-app-id-2 (add-dummy-draft-application! {:date-time (time/minus (time/now) (time/days 120))
+                                                          :actor "alice"})]
+      (let [before-apps (applications/get-all-applications "alice")]
+        (is (= 5 (count before-apps)))
+        (is (every? #{app-id-1 app-id-2 app-id-3 expired-app-id expired-app-id-2}
+                    (mapv :application/id before-apps))))
```

Comment:
We could compare two sets, then it's obvious that they match exactly? Also the `=` comparison usually gives a nice diff on error.